doofinder-for-woocommerce/includes/api/endpoints/class-endpoint-product.php

```diff
@@ -221,32 +221,38 @@ private static function merge_custom_attributes($data, $custom_attr) {
             return isset($attr['type']) && $attr['type'] !== 'metafield';
         }));
 
-        if (!empty($custom_attr)) {
-            $data_with_attr = array_merge($data, self::get_custom_attributes($data["id"], $custom_attr));
+        if (empty($custom_attr)) {
+            return $data;
+        }
 
-            foreach ($custom_attr as $custom) {
-                $attributeKey = $custom["attribute"];
-                $fieldKey     = $custom["field"];
+        $data_with_attr = array_merge($data, self::get_custom_attributes($data["id"], $custom_attr));
 
-                //Exchange renamed fields
-                if (isset($data_with_attr[$attributeKey])) {
-                    $data_with_attr[$fieldKey] = $data_with_attr[$attributeKey];
+        foreach ($custom_attr as $custom) {
+            $attributeKey = $custom["attribute"];
+            $fieldKey     = $custom["field"];
 
-                    //We delete the original key only if it has been renamed to a different alias.
-                    if ($fieldKey != $attributeKey) {
-                        unset($data_with_attr[$attributeKey]);
-                    }
+            if (!isset($data_with_attr[$attributeKey])) {
+                continue;
+            }
 
-                    //List of value options
-                    if (is_array($data_with_attr[$fieldKey])) {
-                        $data_with_attr[$fieldKey] = array_column($data_with_attr[$fieldKey], "name");
-                    }
-                }
+            //Exchange renamed fields
+            $data_with_attr[$fieldKey] = $data_with_attr[$attributeKey];
+
+            //We delete the original key only if it has been renamed to a different alias.
+            if ($fieldKey != $attributeKey) {
+                unset($data_with_attr[$attributeKey]);
             }
 
-            return $data_with_attr;
+            //List of value options
+            if (is_array($data_with_attr[$fieldKey])) {
+                $name_column = array_column($data_with_attr[$fieldKey], "name");
+
+                if (!$name_column) {
+                    $data_with_attr[$fieldKey] = $name_column;
+                }    
```

Comment:
@juan-gm igual me estoy liando pero este caso no sería al revés? o igual es que la comprobación se me hace rara.. $name_column va a tener un array vacío si no encuentra nada y un array con valores si encuentra algo, entonces, el if no debería ser algo como esto?:

```suggestion
                if (!empty($name_column)) {
                    $data_with_attr[$fieldKey] = $name_column;
                }    
```

tal y como lo tienes creo que es al contrario, no? que si no hay $name_column entonces estás sobreescribiendo el valor?

Reply:
cierto lo cambio! Gracias